llmobs/experiment/experiment.go

```diff
@@ -0,0 +1,423 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package experiment
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/DataDog/dd-trace-go/v2/instrumentation/errortrace"
+	illmobs "github.com/DataDog/dd-trace-go/v2/internal/llmobs"
+	"github.com/DataDog/dd-trace-go/v2/internal/llmobs/transport"
+	"github.com/DataDog/dd-trace-go/v2/internal/log"
+	"github.com/DataDog/dd-trace-go/v2/internal/version"
+	"github.com/DataDog/dd-trace-go/v2/llmobs/dataset"
+)
+
+var (
+	errRequiresProjectName = errors.New(`a project name must be provided for the experiment, either configured via the DD_LLMOBS_PROJECT_NAME
+environment variable, using the global tracer.WithLLMObsProjectName option, or experiment.WithProjectName option`)
+	errRequiresAppKey = errors.New(`an app key must be provided for the experiment in agentless mode configured via the DD_APP_KEY environment variable`)
+)
+
+// Experiment represents a DataDog LLM Observability experiment.
+type Experiment struct {
+	Name string
+
+	cfg         *newCfg
+	task        Task
+	dataset     *dataset.Dataset
+	evaluators  []Evaluator
+	description string
+	tagsSlice   []string
+
+	// these are set after the experiment is run
+	id      string
+	runName string
+}
+
+// Task represents the task to run for an Experiment.
+type Task interface {
+	Name() string
+	Run(ctx context.Context, inputData map[string]any, experimentCfg map[string]any) (any, error)
+}
+
+// Evaluator represents an evaluator for an Experiment.
+type Evaluator interface {
+	Name() string
+	Run(ctx context.Context, input map[string]any, output any, expectedOutput any) (any, error)
+}
+
+// TaskFunc is the type for Task functions.
+type TaskFunc func(ctx context.Context, inputData map[string]any, experimentCfg map[string]any) (any, error)
+
+type namedTask struct {
+	name string
+	fn   TaskFunc
+}
+
+func (n *namedTask) Name() string {
+	return n.name
+}
+
+func (n *namedTask) Run(ctx context.Context, inputData map[string]any, experimentCfg map[string]any) (any, error) {
+	return n.fn(ctx, inputData, experimentCfg)
+}
+
+// NewTask creates a new Task.
+func NewTask(name string, fn TaskFunc) Task {
+	return &namedTask{
+		name: name,
+		fn:   fn,
+	}
+}
```

Comment:
@rarguelloF What's the reason to return as interface instead of the underlying `NamedTask` struct? I didn't find any other type implementing `Task`. Are there expectations to add other types of tasks?